### PR TITLE
Slow deepen split

### PR DESCRIPTION
**Pull request by keithbostic**
@michael, I think this one is ready to merge if you agree with the change.

### CODE REVIEW

**Comment by wtbuild** [comment]
Test PASSed.

File: src/btree/bt_split.c [hunk 1]
```diff
@@ -185,38 +183,18 @@ __split_should_deepen(WT_SESSION_IMPL *session, WT_PAGE *page)
...
 	pindex = WT_INTL_INDEX_COPY(page);
 
 	/*
 	 * Deepen the tree if the page's memory footprint is larger than the
-	 * maximum size for a page in memory.  We need an absolute minimum
-	 * number of entries in order to split the page: if there is a single
-	 * huge key, splitting won't help.
+	 * maximum size for a page in memory (presuambly putting eviction
+	 * pressure on the cache). Additionally, ensure the page has enough
+	 * entries to make it worth splkitting, in the case of a set of large
```
Comment (sueloverso) [inline src/btree/bt_split.c]: Typo - splkitting

File: src/btree/bt_split.c [hunk 2]
```diff
@@ -405,8 +383,7 @@ __split_deepen(WT_SESSION_IMPL *session, WT_PAGE *parent)
 	 * Create N children, unless we are dealing with a large page without
 	 * many entries, in which case split into the minimum number of pages.
 	 */
-	children = WT_MAX(pindex->entries / __split_deepen_per_child,
-	    __split_deepen_min_child);
+	children = pindex->entries / __split_deepen_per_child;
```
Comment (michaelcahill) [inline src/btree/bt_split.c]: Why this change: if we had `memory_page_max` set to 1MB and 100KB keys, we could still end up with `children == 0` here?

File: src/btree/bt_split.c [hunk 3]
```diff
@@ -12,10 +12,8 @@
  * Tuning; global variables to allow the binary to be patched, we don't yet have
  * any real understanding of what might be useful to surface to applications.
  */
-static u_int __split_deepen_max_internal_image = 100;
-static u_int __split_deepen_min_child = 10;
-static u_int __split_deepen_per_child = 100;
-static u_int __split_deepen_split_child = 100;
+static u_int __split_deepen_min_child = 10000;
+static u_int __split_deepen_per_child = 1000;
```
Comment (michaelcahill) [inline src/btree/bt_split.c]: Why did the constants change?

File: src/btree/bt_split.c (hunk 1 above)
Comment (keithbostic) [inline src/btree/bt_split.c]: Thanks, fixed.

File: src/btree/bt_split.c (hunk 2 above)
Comment (keithbostic) [inline src/btree/bt_split.c]: The new version of `__split_should_deepen` checks there are at least `__split_deepen_min_child` entries, so this check can be simpler.

File: src/btree/bt_split.c (hunk 3 above)
Comment (keithbostic) [inline src/btree/bt_split.c]: It seems pointless to split a page (even a big page), if there's nothing on it to split.

Comment (michaelcahill) [inline src/btree/bt_split.c]: That wasn't a problem with the previous values.  What's better about the new values?

File: src/btree/bt_split.c (hunk 2 above)
Comment (michaelcahill) [inline src/btree/bt_split.c]: We're not dividing by `__split_deepen_min_child`, we're dividing by a different constant.  It happens to be smaller in the values you've chosen, but the previous code didn't rely on that.

**Comment by wtbuild** [comment]
Test PASSed.

**Comment by michaelcahill** [comment]
Thanks, @keithbostic, this lgtm.  I'll merge.

**Comment by wtbuild** [comment]
Test PASSed.

**Comment by wtbuild** [comment]
Test PASSed.
